### DISCUSSION TITLE
qucs-s (ngspice) vs ltspice or the new qspice

### DISCUSSION THREAD

**Discussion by viticleri** [ra3xdh/qucs_s#545] [Q&A]
Hi,
Can someone explain me the main differences between these 3 simulation software packages?

Thanks in advance.

viticleri

**Comment by ra3xdh** [comment]
Ngspice is open-source and cross-platform. Both LTSpice and QSpice are proprietary and Windows-only. LTSpice and QSpice may use incompatible netlist syntax extensions.

**Comment by viticleri** [comment]
Thks, these are a well known issues but I really want to know about the inner workings of qucs-s in a way of speed, usability, "plugins", documentation, support, learning curve... 


Regards.

**Comment by ra3xdh** [comment]
> I really want to know about the inner workings of qucs-s in a way of speed

The simulation speed of Ngspice is comparable to proprietary simulators. Refer to https://ngspice.sourceforge.io/ 

> usability,

LTSpice/QSpice may use specific netlist syntax extnesions. Something will work using LTspice compatibility mode. Some LTSpice libraries will not with work with Ngspice. Refer to https://ngspice.sourceforge.io/docs/ngspice-html-manual/manual.xhtml#magicparlabel-23998

The usability of Qucs-S and LTspice from the GUI side cannot be directly compared. Everything depends on the user preferences. 

> documentation

Documentation of Qucs-S still be lacking. The writing of the documentation consumes much more time than writing a code. But everyone may contribute. The documentation on readthedocs.io is outdated and need to be 100% rewritten.

> support

Qucs-S is open-source software. The support is not guaranteed. This is valid for every open-source software developed in a spare time.

> learning curve...

You can find even some video tutorials. The new tutorial is a good starting point: https://ra3xdh.github.io/pdf/qucs_s_tutorial.pdf

**Reply by viticleri** [reply]
Thank you very much for your reply!
Which is the best way to help in the "outdated" documentation part?
How to do it?
Where to begin with?
The coordination people?
The guidelines to follow?...

I think if people have a well written documentation could boost up the  qucs-s open source project! 

Regards. Toni

**Reply by ra3xdh** [reply]
> Which is the best way to help in the "outdated" documentation part?

See https://github.com/ra3xdh/qucs_s/issues/554 I have drafted a plan how to rewrite the documentation.

**Comment by ra3xdh** [comment]
> plugins

Qucs-S doesn't support user extensions like browser plugins. LTSpice also doesn't provide plugin API.
